### PR TITLE
Pact 3

### PR DESCRIPTION
**Pull request by fosskers**
This PR bumps our Pact dependency to Pact 3.0.1.

### CODE REVIEW

File: tools/txg/TXG.hs [hunk 1]
```diff
@@ -343,20 +344,18 @@ singleTransaction args host (SingleTX c cid)
...
-    -- TODO Make `ExceptT` once `RequestKeys` is a `NonEmpty`.
-    f :: TXGConfig -> Command Text -> MaybeT IO ListenResponse
-    f cfg@(TXGConfig _ _ ce v _) cmd =
-      hushT (ExceptT $ sendTransactions cfg cid [cmd]) >>= \case
-        RequestKeys [] -> nothing
-        RequestKeys (rk:_) -> hushT . ExceptT $ runClientM (listen v cid $ ListenerRequest rk) ce
+    f :: TXGConfig -> Command Text -> ExceptT ServantError IO ListenResponse
+    f cfg@(TXGConfig _ _ ce v _) cmd = do
+      RequestKeys (rk :| _) <- ExceptT . sendTransactions cfg cid $ pure cmd
+      ExceptT $ runClientM (listen v cid $ ListenerRequest rk) ce
```
Comment (fosskers) [inline tools/txg/TXG.hs]: We got our wish, @emmanueldenloye  :)

**Comment by fosskers** [comment]
Woops, gotta please Nix.

**Comment by emmanueldenloye** [comment]
Let's wait until CI is done before you merge.

**Comment by fosskers** [comment]
@emilypi looks like perhaps the Coin Contract needs to be updated?

**Comment by emilypi** [comment]
@fosskers I pushed 2 changes: 

1. We were missing the `semialign` package to get us `Data.Align` for some tests. I've added it into the dependency list

2. The problem was not in the coin contract, but in the way we were generating `GasPrice`s. The new `GasPrice` parser is restrictive in that it now requires *non-negative Decimals*, and throws when you end up parsing a negative gasprice as part of the `FromJSON` parser - this is something we'll need to harden, as this is a bit of a surprise to me as well. A refinement type would be very much preferred instead of just throwing, but it's low priority since `GasPrice` is only ever set by us. Anyway, I've come up with the correct `Arbitrary GasPrice` instance and tests are good to go.

**Pull request merged**